components/brave_rewards/resources/page/components/pageWallet.tsx

```diff
@@ -433,7 +433,22 @@ class PageWallet extends React.Component<Props, State> {
     return getLocale('greetingsVerified', { name: externalWallet.userName })
   }
 
+  disconnectWalletAlert = (): AlertWallet | null => {
+    const { ui, externalWallet } = this.props.rewardsData
+    const walletType = externalWallet ? externalWallet.type : ""
+    const showDisconnectWalletError = ui && ui.disconnectWalletError
+    if (showDisconnectWalletError) {
+      return {
+        node: <><b>{getLocale('uhOh')}</b><br />{getLocale('disconnectWalletFailed', { wallet: walletType })}</>,
+        type: 'error',
+      }
+    }
+    return null
+  }
+
   onDisconnectClick = () => {
+    const { ui } = this.props.rewardsData
+    ui.disconnectWalletError = false
```

Comment:
We shouldn't mutate `rewardsData` from here - instead, we'll have to make a call through `this.actions.` to update the Redux state. I think maybe this will work?

```js
this.actions.onClearAlert('showDisconnectWalletError')
```